crux-mir/report-coverage/src/main.rs

```diff
@@ -91,11 +95,18 @@ fn parse_report_into(json: Value, r: &mut Report) -> Result<(), String> {
             .ok_or_else(|| format!("callgraph section has no `events` field"))?;
         let events = events.as_array()
             .ok_or_else(|| format!("expected callgraph `events` field to be an array"))?;
+        let mut first_block = true;
         for evt in events {
             let evt = evt.as_object()
                 .ok_or_else(|| format!("expected event to be an object"))?;
             if evt.get("type").map_or(false, |j| j == "BLOCK") {
                 let fn_id = event_function(evt)?;
+                if first_block {
+                    if let Some(c) = get_crate(&fn_id) {
+                        r.report_crates.push(c);
```

Comment:
If I'm reading this code correctly, `report_crates` is a `Vec` that can only ever have at most one element. Would it make more sense for `report_crates` to be an `Option<String>` instead to avoid the unrepresentable state of 2 or more crates?

Reply:
Ah, this is the part I was missing. In that case, a `Vec` seems like a reasonable choice.